### PR TITLE
Single and quadruple precisions for load/savetxt

### PR DESCRIPTION
**Pull request by jvdp1**

### CODE REVIEW

File: src/stdlib_experimental_io.f90 [hunk 1]
```diff
@@ -1,28 +1,65 @@
...
 
 interface savetxt
     module procedure ssavetxt
     module procedure dsavetxt
+    module procedure qsavetxt
 end interface
 
 contains
 
+!PUBLIC
```
Comment (certik) [inline src/stdlib_experimental_io.f90]: What does this `!PUBLIC` do?

File: src/stdlib_experimental_io.f90 [hunk 2]
```diff
@@ -1,28 +1,65 @@
 module stdlib_experimental_io
-use iso_fortran_env, only: sp=>real32, dp=>real64
+use iso_fortran_env, only: sp=>real32, dp=>real64 ,qp=>real128
```
Comment (certik) [inline src/stdlib_experimental_io.f90]: Space after `,` and no space before?

File: src/stdlib_experimental_io.f90 [hunk 3]
```diff
@@ -1,28 +1,65 @@
...
     module procedure ssavetxt
     module procedure dsavetxt
+    module procedure qsavetxt
 end interface
 
 contains
 
+!PUBLIC
 subroutine sloadtxt(filename, d)
+! Loads a 2D array from a text file.
```
Comment (certik) [inline src/stdlib_experimental_io.f90]: For documentation, once we move to use templates as in #35, then the documentation will be written only once, and the templating engine will generate all these `sloadtxt` automatically with the same documentation. Until then, it's fine to copy it by hand.

File: src/stdlib_experimental_io.f90 [hunk 4]
```diff
@@ -50,34 +87,59 @@ subroutine dloadtxt(filename, d)
 !     11 12 13
 !     ...
 !
-character :: c
-integer :: s, ncol, nrow, ios, i
-logical :: lastwhite
-real(dp) :: r
+integer :: s
+integer ::nrow,ncol,i
```
Comment (certik) [inline src/stdlib_experimental_io.f90]: Space after `::` ?

File: src/stdlib_experimental_io.f90 [hunk 5]
```diff
@@ -50,34 +87,59 @@ subroutine dloadtxt(filename, d)
...
-lastwhite = .true.
-do
-   read(s, '(a)', advance='no', iostat=ios) c
-   if (ios /= 0) exit
-   if (lastwhite .and. .not. whitechar(c)) ncol = ncol + 1
-   lastwhite = whitechar(c)
-end do
-
-rewind(s)
+ncol=number_of_columns(s)
```
Comment (certik) [inline src/stdlib_experimental_io.f90]: Space around `=` ?

File: src/stdlib_experimental_io.f90 [hunk 6]
```diff
@@ -115,6 +195,70 @@ subroutine dsavetxt(filename, d)
...
+integer :: s, i
+open(newunit=s, file=filename, status="replace")
+do i = 1, size(d, 1)
+    write(s, *) d(i, :)
+end do
+close(s)
+end subroutine
+
+
+!PRIVATE
```
Comment (certik) [inline src/stdlib_experimental_io.f90]: I assume this does a similar thing as the `!PUBLIC` above.

File: src/stdlib_experimental_io.f90 (hunk 1 above)
Comment (jvdp1) [reply src/stdlib_experimental_io.f90]: I am used to sort the different procedures in groups of public and private procedures. This `!PUBLIC` (and `!PRIVATE` below) refer to these two groups. I can remove them.

File: src/stdlib_experimental_io.f90 (hunk 6 above)
Comment (jvdp1) [reply src/stdlib_experimental_io.f90]: Removed

File: src/stdlib_experimental_io.f90 (hunk 5 above)
Comment (jvdp1) [reply src/stdlib_experimental_io.f90]: Corrected

File: src/stdlib_experimental_io.f90 (hunk 4 above)
Comment (jvdp1) [reply src/stdlib_experimental_io.f90]: Corrected

File: src/stdlib_experimental_io.f90 (hunk 2 above)
Comment (jvdp1) [reply src/stdlib_experimental_io.f90]: Corrected

File: src/stdlib_experimental_io.f90 (hunk 1 above)
Comment (jvdp1) [reply src/stdlib_experimental_io.f90]: Now removed.

**Comment by certik** [comment]
This is +1 from me to merge as is. The changes are straightforward, they do not change the public API and they all happen in experimental, so I am going to merge it.

**Comment by certik** [comment]
@jvdp1 thanks for the contribution!

File: src/stdlib_experimental_io.f90 (hunk 1 above)
Comment (certik) [reply src/stdlib_experimental_io.f90]: I see. Once we fix #4, we'll consolidate all such things.

**Comment by fiolj** [comment]
I've mentioned in the issue on quadrature that I'd like to have implementations for integrating complex functions. I think, that seeing as complex numbers are implemented in fortran, we could include them in the library. I've made the modifications to common.fypp, stdlib_experimental_io.fypp, and test_savetxt.f90 to add support for complex numbers. Is that desired? Should I open a pull request?

**Comment by jvdp1** [comment]
@fiolj same comment as in [#62 comment](https://github.com/fortran-lang/stdlib/issues/62#issuecomment-581108893): I think it is a good idea to add `complex` numbers to the library. Because it is an addition to already existing API, I suggest to open a PR.
Could you also add them to `sdtlib_experimental_stat`, please? So the whole library would support `complex` numbers. The community can then further discuss its utility/wish around a PR.

**Comment by fiolj** [comment]
Yes @jvdp1, I'll try this week to put in stats, probably not before tuesday or wednesday. When I was to do it I realized that I should start with `io`
I just put the pull request to `stdlib_experimental_io`, and modified optval, I'll put this also in stats.

**Pull request merged**